app/components/pagination.hbs

```diff
@@ -8,17 +8,25 @@
     </LinkTo>
   </span>
 {{/if}}
+
 {{#each (range 1 (inc @meta.total_pages)) as |number|}}
-  {{#if (eq number @meta.page)}}
-    {{number}}
-  {{else}}
-    <span class="pagination">
-      <LinkTo @route={{@route}} @query={{hash page=number}}>
-        {{number}}
-      </LinkTo>
-    </span>
+  {{#if (or (and (lte number (inc 4 @meta.page))
+                 (gte number (dec 5 @meta.page)))
+            (and (lte @meta.page 5) (lte number 10))
+            (and (gt @meta.page (dec 4 @meta.total_pages))
+                 (gte number (dec 9 @meta.total_pages))))}}
```

Comment:
Thanks again for your contribution. I tested this and it seems to work alright, however I'm having a hard time understanding what is happening here :sweat_smile: 
If there were variable names instead of hard coded numbers, I think it would help a lot to make this easier to read.

Is there some way we could simplify this code, and perhaps extract it to a helper? Let me know what you think.

Reply:
It's basically maintaining a sliding window of length 10, in the range [page - 5, page + 4], with some special handling near the beginning when page <= 5 and near the end when page > total_pages - 4 to avoid out of bounds error. I hope that clears things up a little bit.

I agree hardcoding these magic numbers is against best practices, but I am kind of new to Ember.js and Handlebars, so I am not sure where to go about refactoring this. Might worth reviewing https://github.com/datafruits/datafruits/pull/129.

Reply:
The logic seems fine.

Basically, it's best to keep as much logic out of the templates as possible. You should be able to move all your logic into a helper function. You can learn how to make one here:
https://guides.emberjs.com/release/components/helper-functions/

Reply:
Thanks for the pointer. I'll take a look when I have time.